File: 66-PlusOne/cpp/solution.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        if (digits[digits.size() - 1] != 9) {
            digits[digits.size() - 1]++;
            return digits;
        } 
        else if (digits[digits.size() - 1] == 9) {
            for (int i = digits.size() - 1; i >= 0; i--) {
                if (digits[i] == 9) {
                    digits[i] = 0;
                } else {
                    digits[i]++;
                    break;
                }
            }
            if (digits[0] == 0) {
                digits.insert(digits.begin(), 1);
            }
        }
        return digits;
    }
};

File: 66-PlusOne/javascript/solution.js
// Last updated: 3/6/2026, 12:46:12 AM
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
        if(digits[digits.length - 1] != 9) {
        digits[digits.length - 1]++
    }
    else if(digits[digits.length - 1] == 9) {
        for(let i = digits.length - 1; i >= 0; i--) {
            if(digits[i] == 9) {
                digits[i] = 0
            }
            else {
                digits[i]++
                break
            }
        }
        if(digits[0]==0){
            digits.unshift(1)
        }
    }
    return digits
};